src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -1314,6 +1314,16 @@ public static native byte[] createOperationalCertificate(
    */
   public native byte[] extractSkidFromPaaCert(byte[] paaCert);
 
+  public static native byte[] extractAkidFromPaiCert(byte[] paiCert);
+
+  public static native void validateAttestationInfo(
```

Comment:
- If this is static, it should be in a new interface + implementation class, not in ChipDeviceController.java directly.
- extractSkidFromPaaCert/extractAkidFromPaiCert should be in a new certificate utilities interface/class, and don't belong here.
- Attestation checks require the attestation challenge, which is not passed here.
- Suggest removing PAA cert from here, and let the implementation of device attestation checks use a method of its choosing to access the PAA set. This will allow better control over revocation, and avoid the caller needing to do a lookup, which, if the attestation check was done over cloud, could be based on a more dynamic PAA list than PAA trust store.
- This method doesn't take a callback, and doesn't return a value. How is the management done of the operation?

Reply:
- If this is static, it should be in a new interface + implementation class, not in ChipDeviceController.java directly.
- extractSkidFromPaaCert/extractAkidFromPaiCert should be in a new certificate utilities interface/class, and don't belong here.
  -> Yes, I can create a separate class. Would it be appropriate to also move the existing “extractSkidFromPaaCert” method?

- Attestation checks require the attestation challenge, which is not passed here.
- Suggest removing PAA cert from here, and let the implementation of device attestation checks use a method of its choosing to access the PAA set. This will allow better control over revocation, and avoid the caller needing to do a lookup, which, if the attestation check was done over cloud, could be based on a more dynamic PAA list than PAA trust store.
  -> I agree with you. I think it would be better to use AttestationTrustStore, but the problem is occurring because it returns a paa bytearray rather than a callback. Do you know a good solution?

- This method doesn't take a callback, and doesn't return a value. How is the management done of the operation?
 -> Currently, it is implemented to generate an exception if verification fails.

Reply:
>-> Yes, I can create a separate class. Would it be appropriate to also move the existing “extractSkidFromPaaCert” method?

Yes

> -> I agree with you. I think it would be better to use AttestationTrustStore, but the problem is occurring because it returns a paa bytearray rather than a callback. Do you know a good solution?

The AttestationTrustStore is an implementation detail that tries to get the associated PAA by given SKI. The issue is that the AttestationTrustStore is called "too early" here --> The validateAttestationInfo should only take inputs from the commissionee (currently untrusted elements) and then do all the necessary internal calls to actually determine if PAA is available. This will also help roll-in revocation support later, which will require even more complex checks against  databases that are not directly  a replica of the DCL.

> - This method doesn't take a callback, and doesn't return a value. How is the management done of the operation?
-> Currently, it is implemented to generate an exception if verification fails.

What about on success?

Reply:
> The AttestationTrustStore is an implementation detail that tries to get the associated PAA by given SKI. The issue is that the AttestationTrustStore is called "too early" here --> The validateAttestationInfo should only take inputs from the commissionee (currently untrusted elements) and then do all the necessary internal calls to actually determine if PAA is available. This will also help roll-in revocation support later, which will require even more complex checks against databases that are not directly a replica of the DCL.

-> Yes, all processes verified by DefaultDACVerifier have been added to the currently implemented method.

> This method doesn't take a callback, and doesn't return a value. How is the management done of the operation?
> -> Currently, it is implemented to generate an exception if verification fails.
> What about success?

-> Moves to the next line without any action.

Reply:
@tcarmelveilleux 
I modified in seperate class. please review this.